src/primitives.ts

```diff
@@ -220,6 +220,44 @@ function split<T extends string, D extends string = ''>(str: T, delimiter?: D) {
   return str.split(delimiter ?? ('' as const)) as Split<T, D>
 }
 
+/**
+ * Checks if a string starts with another string.
+ * T: The string to check.
+ * S: The string to check against.
+ * P: The position to start the search.
+ */
+type StartsWith<
+  T extends string,
+  S extends string,
+  P extends number = 0,
+> = Math.IsPositive<P> extends true
+  ? P extends 0
+    ? T extends `${infer TFirst}${infer TRest}`
+      ? S extends `${infer SFirst}${infer SRest}`
+        ? TFirst extends SFirst
+          ? StartsWith<TRest, SRest, P> // Compare next character
+          : false // T differs from S
+        : true // S (Search) is exhausted (and didn't fail)
+      : false // T (Text) is exhausted with S (Search) leftover
+    : StartsWith<Slice<T, P>, S, 0> // P is >0, slice
```

Comment:
```suggestion
  ? Slice<T, P> extends `${S}${string}` ? true : false
    : StartsWith<Slice<T, P>, S, 0> // P is >0, slice
```

Reply:
Oooooh I forgot that TS could do 
```ts
`${S}${string}`
```

Reply:
Your suggestion won't work as-is because of the way `startsWith` handles negative position, but with a little bit of adaptation I can replace the existing type with something a bit shorter